atlasaction/action.go

```diff
@@ -911,6 +910,126 @@ func (a *Actions) SchemaApply(ctx context.Context) error {
 	return nil
 }
 
+// schemaApplyWithApproval applies schema changes with an approval process.
+// It waits for the schema plan to be approved before applying the changes.
+func (a *Actions) schemaApplyWithApproval(ctx context.Context) ([]*atlasexec.SchemaApply, error) {
+	tc, err := a.GetTriggerContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// Extract repo name from the URL it's provided. e.g. "atlas://repo-name?tag=tag-name"
+	var repo string
+	u, err := url.Parse(a.GetInput("to"))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse the URL input: %w", err)
```

Comment:
This leaks the URL (which might contain sensitive info) to the logs in case it is malformed.